src/plugins/analysis/users_and_passwords/code/password_file_analyzer.py

```diff
@@ -112,9 +112,16 @@ def crack_hash(passwd_entry: bytes, result_entry: dict, format_term: str = '') -
     with NamedTemporaryFile() as fp:
         fp.write(passwd_entry)
         fp.seek(0)
-        command = f'{JOHN_PATH} --wordlist={WORDLIST_PATH} {fp.name} {format_term}'
-        result_entry['log'] = execute_shell_command(command)
-        output = execute_shell_command(f'{JOHN_PATH} {fp.name} --show {format_term}').split('\n')
+        john_process = subprocess.run(
+            f'{JOHN_PATH} --wordlist={WORDLIST_PATH} {fp.name} {format_term}',
+            shell=True,
+            stdout=PIPE,
+            stderr=STDOUT,
+            universal_newlines=True,
+        )
+        result_entry['log'] = john_process.stdout
+        john_process = subprocess.run(f'{JOHN_PATH} {fp.name} --show {format_term}', shell=True, stdout=PIPE, stderr=STDOUT, universal_newlines=True)
+        output = john_process.stdout.split('\n')
```

Comment:
again I'm not 100% sure this will not trigger a `UnicodeDecodeError` and it would be safer to do the `decode()` manually with `errors='replace'`

Reply:
I reconsidered this and since the passwords that are in the word list probably won't contain undecodable password strings, this also should not be an issue,